src/tezos_rpc/fetch_block_header.ml

```diff
@@ -0,0 +1,23 @@
+open Tezos
+open Http
+
+type response = Block_header.t
+
+let path ~chain ~block_hash =
+  (* TODO: I don't like this Format.sprintf *)
+  Format.sprintf "/chains/%s/blocks/%s/header" chain block_hash
+
+let execute ~node_uri ~chain ~block_hash =
+  let chain =
+    match chain with
+    | Some chain -> Chain_id.to_string chain
```

Comment:
Stupid question, does this "business" belongs to this function? Why don't change `chain_id.t` to a required field, and give the responsibility of giving `main` to the caller?
Please note, once again, I have no clue of the business behind this 😅

Reply:
Chain_id.t cannot be main, this is a special property of this endpoint, it actually accepts `chain_id | "main"`, we could make a variant for that, and I'm still thinking over it, but the easier solution was to do `chain_id option`

Reply:
Either I don't understand your answer, or you didn't understand my question 😅.
Here, in `execute` function, we do some basic `match` condition on the `~chain`. If `execute` had been called without `chain` (since it's optional) it will use `main`. Right?

My question is:
Should this function `execute` be responsible of this? In fine, it will be exactly the same. But having `execute ~node_uri chain ~block_hash` can be totally different of having `execute ~node_uri ~chain ~block_hash`.

In my mind, I  want, as fair as possible, my functions to be "functionally scoped". By that, I mean I don't want them to calculate things they are not responsible for.

Reply:
Finally got it! Sorry...